cdap-docs/_common/common-build.sh

```diff
@@ -218,15 +218,22 @@ function build_extras() {
 }
 
 function set_mvn_environment() {
+  check_build_rst
   cd ${PROJECT_PATH}
   if [[ "${OSTYPE}" == "darwin"* ]]; then
     # TODO: hard-coded Java version 1.7
     export JAVA_HOME=$(/usr/libexec/java_home -v 1.7)
   fi
+}
+
+function check_build_rst() {
+  local current_directory=$(pwd)
+  cd ${PROJECT_PATH}
   # check BUILD.rst for changes
   BUILD_RST_PATH="${PROJECT_PATH}${BUILD_RST}"
   test_an_include "${BUILD_RST_HASH}" "${BUILD_RST_PATH}"
   echo
+  cd ${current_directory}
```

Comment:
Yeah, that makes sense then. Sub-shells would solve that problem, but by adding complexity in that variables would need to be exported and then results recovered, which might be do-able in that I do use a temp file for holding message results...something for me to keep in mind in the future.